contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/util/SimpleHttp.java

```diff
@@ -0,0 +1,235 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http.util;
+
+import okhttp3.Cache;
+import okhttp3.Credentials;
+import okhttp3.FormBody;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.OkHttpClient.Builder;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.exec.ExecConstants;
+import org.apache.drill.exec.ops.FragmentContext;
+import org.apache.drill.exec.store.http.HttpAPIConfig;
+import org.apache.drill.exec.store.http.HttpStoragePluginConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+
+/**
+ * This class performs the actual HTTP requests for the HTTP Storage Plugin. The core method is the getInputStream()
+ * method which accepts a url and opens an InputStream with that URL's contents.
+ */
+public class SimpleHttp {
+  private static final Logger logger = LoggerFactory.getLogger(SimpleHttp.class);
+
+  private final OkHttpClient client;
+
+  private final HttpStoragePluginConfig config;
+
+  private final FragmentContext context;
+
+  private final HttpAPIConfig apiConfig;
+
+  public SimpleHttp(HttpStoragePluginConfig config, FragmentContext context, String connectionName) {
+    this.config = config;
+    this.context = context;
+    this.apiConfig = config.connections().get(connectionName);
+    client = setupServer();
+  }
+
+
+
+  public InputStream getInputStream(String urlStr) {
+    Request.Builder requestBuilder;
+
+    // The configuration does not allow for any other request types other than POST and GET.
+    if (apiConfig.method().equals("get")) {
+      // Handle GET requests
+      requestBuilder = new Request.Builder().url(urlStr);
+    } else {
+      // Handle POST requests
+      FormBody.Builder formBodyBuilder = buildPostBody();
+      requestBuilder = new Request.Builder()
+        .url(urlStr)
+        .post(formBodyBuilder.build());
+    }
+
+    // Add headers to request
+    if (apiConfig.headers() != null) {
+      for (Map.Entry<String, String> entry : apiConfig.headers().entrySet()) {
+        String key = entry.getKey();
+        String value = entry.getValue();
+        requestBuilder.addHeader(key, value);
+      }
+    }
+
+    // Build the request object
+    Request request = requestBuilder.build();
+    logger.debug("Headers: {}", request.headers());
+
+    try {
+      // Execute the request
+      Response response = client
+        .newCall(request)
+        .execute();
+      logger.debug(response.toString());

```

Comment:
You probably *really* don't want to do this. Converting the response to a string causes the client to download the **entire** response to memory, then convert it to a Java string. Doing so completely negates the disk-based buffering that seems to be implemented below.

If you want the string, wrap the call in an `if`-statement, realizing what will happen:

```
if (logger.isDebugEnabled()) {
  // Warning: Will download entire response to memory!
  logger.debug(response.toString());
}
```

Reply:
I'll just remove it.  I was trying to see if the thing worked at all and was actually making HTTP requests.